modules/internal/ChapelLocale.chpl

```diff
@@ -247,6 +247,35 @@ module ChapelLocale {
       return "";
     }
 
+    //
+    // Support for different types of memory:
+    // large, random-access (low latency), and streaming (high bandwidth)
+    // Inspired by madvise()'s support for RANDOM and SEQUENTIAL access
+    //
+    pragma "no doc"
+    proc defaultMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc largeMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc randomMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc streamingMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
```

Comment:
This is something we discussed in a KNL meeting at some point. Notes can be found at chapel-docs/notes/topics/KNL/2016-04-21.txt